### PR TITLE
Design doc: update workflow fetch API

### PR DESCRIPTION
**Pull request by Yancey0623**
related #1873

### CODE REVIEW

File: doc/design/argo_workflow_on_sqlflow.md [hunk 1]
```diff
@@ -101,69 +120,58 @@ where
...
-      message Logs {
-        repeated string content = 1;
+      message Responses {
+        repeated Responses responses = 1;
       }
-      FetchRequest new_request = 1;
-      Logs logs = 2;
+      FetchRequest updated_fetch_since = 2;
+      bool eof = 2;
+      Logs logs = 3;
```
Comment (tonyyang-svail) [inline doc/design/argo_workflow_on_sqlflow.md]: Should `Logs` be part of `Responses`?

File: doc/design/argo_workflow_on_sqlflow.md [hunk 2]
```diff
@@ -101,69 +120,58 @@ where
...
+          }
         }
-      }
 
-      return &FetchRequest{
-        job             : req.Job,
-        stepId          : nextStep,
-        logOffset       : newLogOffset,
-        finishFetching  : finishFetching,
+        log, newOffset = k9s.ReadPodLogs(pod.Name, offset)
```
Comment (tonyyang-svail) [inline doc/design/argo_workflow_on_sqlflow.md]: typo: k9s -> k8s

**Review by tonyyang-svail — DISMISSED** [review]
LGTM. @weiguoz please also take a look at this.

File: doc/design/argo_workflow_on_sqlflow.md (hunk 1 above)
Comment (weiguoz) [inline doc/design/argo_workflow_on_sqlflow.md]: Where is the definition of `Logs`?

File: doc/design/argo_workflow_on_sqlflow.md [hunk 3]
```diff
@@ -101,69 +120,58 @@ where
...
+      string step_id = 2;
+      string step_phase = 3;
     }
 
     message FetchResponse {
-      message Logs {
-        repeated string content = 1;
+      message Responses {
+        repeated Responses responses = 1;
       }
```
Comment (weiguoz) [inline doc/design/argo_workflow_on_sqlflow.md:144]: Should `Responses` be singular?

File: doc/design/argo_workflow_on_sqlflow.md (hunk 1 above)
Comment (Yancey0623) [reply doc/design/argo_workflow_on_sqlflow.md]: I think we can remove it.

Comment (Yancey0623) [reply doc/design/argo_workflow_on_sqlflow.md]: It's deprecated, will remove it.

**Review by tonyyang-svail — DISMISSED** [review]

File: doc/design/argo_workflow_on_sqlflow.md [hunk 4]
```diff
@@ -81,16 +81,35 @@ We implement the `Fetch` API as the following pseudo-code:
 ``` go
 func Fetch(req *FetchRequest) *FetchResponse {
   // kubectl get pod ...
-  pod := getPod(req)
+  step := getStep(req)
```
Comment (typhoonzero) [inline doc/design/argo_workflow_on_sqlflow.md]: `getStep` need `workflow` object to run.

File: doc/design/argo_workflow_on_sqlflow.md [hunk 5]
```diff
@@ -81,16 +81,35 @@ We implement the `Fetch` API as the following pseudo-code:
...
+  stepPhaseLogs := fetchStepPhaseLogs(wf, step)
+
+  // append step phase status logs to Response
+  responses := []pb.Response{&Message{stepPhaseLogs}}
+
+  eof := false // True for end of job
+
+  if podComplete(pod) {
+    // if the current step execute a standard SQL, the query result would be output
+    // as protobuf message string format with prefix SQLFLOW_PROTOBUF:
```
Comment (typhoonzero) [inline doc/design/argo_workflow_on_sqlflow.md:102]: is the serialized message using `pbtxt`?

Comment (Yancey0623) [reply doc/design/argo_workflow_on_sqlflow.md:102]: The comment used `Unmarshl` instead of `UnmarshlText`, I can do more tests when I implement it.

**Review by typhoonzero — APPROVED** [review]

**Pull request merged**